CustomerData/CustomSection.php

```diff
@@ -0,0 +1,145 @@
+        RequestFactory $requestFactory,
+        BalancepayConfig $balancepayConfig,
+        BalancepayHelper $balancepayHelper
+    ) {
+        $this->customerSession = $customerSession;
+        $this->pricingHelper = $pricingHelper;
+        $this->customerRepositoryInterface = $customerRepositoryInterface;
+        $this->requestFactory = $requestFactory;
+        $this->balancepayConfig = $balancepayConfig;
+        $this->balancepayHelper = $balancepayHelper;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getSectionData()
+    {
+        $status = false;
+        $showButton = true;
+        $showCredit = false;
+        $creditLimit = '$0.00';
+        $buyerResponse = $this->getBuyerAmount();
+        if (!empty($buyerResponse['qualificationStatus']) && $buyerResponse['qualificationStatus'] == 'completed'){
+            $status = true;
+            $showButton = false;
+            $showCredit = true;
+            $creditLimit =  $this->formattedAmount($buyerResponse['qualification']['creditLimit'] ?? 0);
+        }
+        return [
+            'creditLimit' => 'Terms Limit: '.$creditLimit,
+            'status' => $status, // $buyerResponse['qualificationStatus'] ?? ''
+            'showButton' => $showButton,
+            'showCredit' => $showCredit
+        ];
+    }
+
+    public function getBuyerAmount()
+    {
+        $response = [];
+        try {
+            $buyerId = $this->getCustomerSessionBuyerId();
+            if (!empty($buyerId)) {
+                $response = $this->requestFactory
+                    ->create(RequestFactory::BUYER_REQUEST_METHOD)
+                    ->setRequestMethod('buyers/' . $buyerId)
+                    ->setTopic('getbuyers')
+                    ->process();
+            }
+        } catch (\Exception $e) {
+            $this->balancepayConfig->log('Get Buyer [Exception: ' .
+                $e->getMessage() . "]\n" . $e->getTraceAsString(), 'error');
+        }
+        return $response;
+    }
+
+    /**
+     * GetCustomerSessionBuyerId
+     *
+     * @return mixed
+     */
+    public function getCustomerSessionBuyerId()
+    {
+        if ($this->customerSession->getBuyerId()) {
+            return $this->customerSession->getBuyerId();
+        }
+        $customerId = $this->balancepayHelper->getCustomerSessionId();
+        if (!empty($customerId)) {
+            $customer = $this->customerRepositoryInterface->getById($customerId);
+            $customerAttributeData = $customer->__toArray();
+            $buyerId = isset($customerAttributeData['custom_attributes']['buyer_id']) ?
+                $customerAttributeData['custom_attributes']['buyer_id']['value'] : '';
+            $this->customerSession->setBuyerId($buyerId);
+            return $buyerId;
+        }
+        return 0;
+    }
+
+    /**
+     * @param $price
+     * @return float|string
+     */
+    public function formattedAmount($price)
```

Comment:
is this method used outside this class? if not consider to make private / protected

Reply:
In ME-29 PR this class has been updated and functions are removed for refactoring